Homework01/task1_4/src/by/bsu/tat/main/Discriminant.java

```diff
@@ -0,0 +1,62 @@
+package by.bsu.tat.main;
+
+/**
+ * This class does finding the roots of a quadratic equation
+ * @author Alexey Makovskiy
+ */
+
+import java.util.Scanner;
+
+/**
+ * This method finds the square roots.
+ * Checks on the correct introduction of the coefficients.
+ * Checks on the discriminant is less than zero
+ */
+
+public class Discriminant {
+    public static void main(String[] args){
+        try {
+            double a, c, d, b;
+            String operation;
+            Scanner sr = new Scanner(System.in);
+            System.out.println("Enter a = ");
+            a = sr.nextDouble();
+            System.out.println("Enter b = ");
+            b= sr.nextDouble();
+            System.out.println("Enter c = ");
+            c = sr.nextDouble();
+            solve (a,b,c);
+            d = Math.pow(b, 2)-4*a*c;
+            if (d == 0) {
+        		System.out.println("x1,2 = "+(-b + Math.sqrt(d))/(2 * a));
+            } else {
+                if (d>0) {
+                    System.out.println("x1 = " + (-b + Math.sqrt(d))/(2 * a));
+                    System.out.println("x2 = " + (-b - Math.sqrt(d))/(2 * a));
+                } else {
+                    System.out.println("Error! Discriminant is less than zero!");
+                }
+            }
+        } catch(Exception e) {
+            System.out.println("You entered not a number!");
+        }
+    }
+
+    /**
+     * In this method, there is a check on the maximum and minimum input number
+     */
+    public static void solve(double a, double b, double c){
+        if (Double.MIN_VALUE > a || a > Double.MAX_VALUE){
+            System.out.println("Warring! You have entered a very large number!");
+            System.exit(0);
```

Comment:
Do not exit application from method or at least this behaviour must be documented